consensus/polybft/consensus_runtime.go

```diff
@@ -996,6 +998,9 @@ func (c *consensusRuntime) IsValidBlock(proposal []byte) bool {
 }
 
 func (c *consensusRuntime) IsValidSender(msg *proto.Message) bool {
+	c.lock.RLock()
```

Comment:
I suggest doing on the top:
```
c.lock.RLock()
fsm := c.fsm
c.lock.RUnlock()
```
Or even better make `func getFsm` which will call every other function that needs fsm -> on the top
`fsm := c.getFsm()`

Reply:
getFSM() should do deep copying then. I believe it's a pointer type so just returning won't help much.

Reply:
@igorcrevar